src/Controls/tests/UITests/Tests/AppiumPlatformsTestBase.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System.Reflection;
+using Microsoft.Maui.Appium;
+using NUnit.Framework;
+using NUnit.Framework.Interfaces;
+using TestUtils.Appium.UITests;
+
+namespace Microsoft.Maui.AppiumTests
+{
+#if ANDROID
+	[TestFixture(TestDevice.Android)]
+#elif IOSUITEST
+	[TestFixture(TestDevice.iOS)]
+#elif MACUITEST
+	[TestFixture(TestDevice.Mac)]
+#elif WINTEST
+	[TestFixture(TestDevice.Windows)]
+#endif
+	public class AppiumPlatformsTestBase : AppiumUITestBase
+	{
+		TestDevice _testDevice;
+		public AppiumPlatformsTestBase(TestDevice device)
+		{
+			_testDevice = device;
+		}
+
+		[TearDown]
+		public void TearDown()
+		{
+			var testOutcome = TestContext.CurrentContext.Result.Outcome;
+			if (testOutcome == ResultState.Error ||
+				testOutcome == ResultState.Failure)
+			{
+				var logDir = (Path.GetDirectoryName(Environment.GetEnvironmentVariable("APPIUM_LOG_FILE")) ?? Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location))!;
+
+				var pageSource = Driver?.PageSource!;
+				File.WriteAllText(Path.Combine(logDir, $"{TestContext.CurrentContext.Test.MethodName}-PageSource.txt"), pageSource);
```

Comment:
Currently only tested on android: 

I think the logDir here always evaluates to `/` because when the script runs it sets the appium_log_dir to

Setting environment variable: APPIUM_LOG_FILE = '/appium_windows.log'

Should we make sure this gets set to an absolute path from the cake script?